app/resources/admin.html.template

```diff
@@ -68,38 +68,99 @@
         continue;
       }
-      if (arr_contains(used_languages, language_code)) {
-        alert("You entered the same language (" + language_code + ") twice. " +
-              "That's not allowed.");
-        return false;
-      }
+      language_code_index_map[language_code] = dropdown_index;
       used_languages.push(language_code);
-      languages_to_titles[language_code] = language_title;
     }
 
     // Copy the dropdowns and the titles into the text boxes that will
     // actually be POSTed (language_menu_options and repo_titles).
     $('language_menu_options').value = JSON.stringify(used_languages);
-    $('repo_titles').value = JSON.stringify(languages_to_titles);
 
+    var repo_titles = {};
+    for (var language_code in language_code_index_map) {
+      var index = language_code_index_map[language_code];
+      var language_value = $('repo_titles_language_value_' + index).value
+      repo_titles[language_code] = language_value;
+    }
+    $('repo_titles').value = JSON.stringify(repo_titles);
+
+    for (var id_num = 0; id_num < MULTILINGUAL_TABLE_IDS.length; id_num++) {
+      var custom_messages = {};
+      var table_id = MULTILINGUAL_TABLE_IDS[id_num];
+      for (var language_code in language_code_index_map) {
+        var index = language_code_index_map[language_code];
+        var language_value = $(table_id + '_language_value_' + index).value;
+        custom_messages[language_code] = language_value;
+      }
+      $(table_id).value = JSON.stringify(custom_messages);
+    }
     return true;
   }
 
+  function get_selected_language(row_number) {
+    var selected_language_codes = 
+        $('repo_titles_language_selection_' + row_number);
+    var selected_index = selected_language_codes.selectedIndex;
+    var selected_language = selected_language_codes[selected_index].value;
+    return selected_language;
+  }
+
+  function validate_selected_languages(row_number) {
+    var used_languages = {};
+    var language_num = $('repo_titles_table').rows.length;
+    var selected_language = get_selected_language(row_number);
+    for (var index = 1; index < language_num; index++) {
+      var language_code = get_selected_language(index);
+      if (language_code != 'unused' && used_languages[language_code] != null) {
+        alert('You entered the same language!!');
+      }
+      used_languages[language_code] = 1;
+    }
+  }
+
+  function escape_html(str) {
+    var escape_table = {
+      '<': '&lt;',
+      '>': '&gt;',
+      '&': '&amp;',
+      '"': '&quot;',
+      '\'': '&#x27;',
+    };
+    var escaped_str = str.replace(/[<>&"\']/g, function (match) {
+      return escape_table[match];
+    });
+    return escaped_str;
+  }
+
+  function update_language_selected(row_number) {
+    var language_codes = $('repo_titles_language_selection_' + row_number);
+    var selected_index = language_codes.selectedIndex;
+    var selected_language = language_codes[selected_index].text;
```

Comment:
get_selected_language() returns language_code.
in update_language_selected(), selected_language = text(repo_titles_language_selected)